dist/aframe-altspace-component.js

```diff
@@ -465,86 +467,744 @@
+					console.error('Entities cannot be synced using link mode without an id.');
+					return;
+				}
+
+				link(syncSys.sceneRef.child(id));
+				setupReceive();
+			}
+
+			function link(entityRef) {
+				ref = entityRef;
+				key = ref.key();
+				dataRef = ref.child('data');
+				component.dataRef = dataRef;
+				ownerRef = ref.child('owner');
+			}
+
+			function setupReceive() {
+
+				//if nobody has owned the object yet, we will. 
+				ownerRef.transaction(function (owner) {
+					if (owner) return undefined;
+
+					ownerRef.onDisconnect().set(null);
+					return syncSys.clientId;
+				});
+
+				ownerRef.on('value',
+					function(snapshot) {
+						var newOwnerId = snapshot.val();
+
+						var gained = newOwnerId === syncSys.clientId && !isMine;
+						if (gained) component.el.emit('ownershipgained', null, false);
+
+
+						//note this also fires when we start up without ownership
+						var lost = newOwnerId !== syncSys.clientId && isMine;
+						if (lost){
+							component.el.emit('ownershiplost', null, false);
+
+							//we no longer have to clear our ownership when we disconnect
+							ownerRef.onDisconnect().cancel();
+						}
+
+						ownerId = newOwnerId;
+
+						isMine = newOwnerId === syncSys.clientId;
+					});
+			}
+
+			this.takeOwnership = function() {
+				console.log('taking ownership of: ' + component.el.id)
+				ownerRef.set(syncSys.clientId);
+
+				//clear our ownership if we disconnect
+				//this is needed if we are the last user in the room, but we expect people to join later
+				ownerRef.onDisconnect().set(null);
+			}
+
+			Object.defineProperty(component, 'isMine', {
+				get: function () {
+					return isMine;//TODO: Should this be state instead?
+				}
+			});
+		}
+	});
+
+	AFRAME.registerComponent('sync-transform',
+	{
+		dependencies: ['sync'],
+		schema: {
+		},
+		init: function () {
+			var sync = this.el.components.sync;
+			var component = this;
+
+			var positionRef = sync.dataRef.child('position');
+			var rotationRef = sync.dataRef.child('rotation');
+			var scaleRef = sync.dataRef.child('scale');
+
+			this.updateRate = 100;
+
+			var stoppedAnimations = [];
+			//pause all animations on ownership loss
+			this.el.addEventListener('ownershiplost', function() {
+				var children = component.el.children;
+				for (var i = 0; i < children.length; i++) {
+					var tagName = children[i].tagName.toLowerCase();
+					if (tagName === "a-animation") {
+						stoppedAnimations.push(children[i]);
+						children[i].stop();
+					}
+				}
+			});
+			this.el.addEventListener('ownershipgained', function () {
+				for (var i = 0; i < stoppedAnimations.length; i++) {
+					var animation = stoppedAnimations[i];
+					animation.start();
+				}
+				stoppedAnimations = [];
+			});
+
+			positionRef.on('value', function (snapshot) {
+				if (sync.isMine) return;
+
+				var value = snapshot.val();
+				if (!value) return;
+
+				component.el.setAttribute('position', {x: value.x, y: value.y, z: value.z});
```

Comment:
Hmm, is it necessary to create a new anon object? Can't you just use `value` directly?